samples/NativeCatalog/ios/NativeCatalog/Examples/CustomPdfView.m

```diff
@@ -0,0 +1,195 @@
+    self.topController = self.pdfController;
+
+  } else {
+    self.topController = [[PSPDFNavigationController alloc] initWithRootViewController:self.pdfController];;
+  }
+
+  UIView *topControllerView = self.topController.view;
+  topControllerView.translatesAutoresizingMaskIntoConstraints = NO;
+
+  [self addSubview:topControllerView];
+  [controller addChildViewController:self.topController];
+  [self.topController didMoveToParentViewController:controller];
+
+  [NSLayoutConstraint activateConstraints:
+   @[[topControllerView.topAnchor constraintEqualToAnchor:self.topAnchor],
+     [topControllerView.bottomAnchor constraintEqualToAnchor:self.bottomAnchor],
+     [topControllerView.leadingAnchor constraintEqualToAnchor:self.leadingAnchor],
+     [topControllerView.trailingAnchor constraintEqualToAnchor:self.trailingAnchor],
+   ]];
+}
+
+- (void)destroyViewControllerRelationship {
+  if (self.topController.parentViewController) {
+    [self.topController willMoveToParentViewController:nil];
+    [self.topController removeFromParentViewController];
+  }
+}
+
+- (UIViewController *)pspdf_parentViewController {
+  UIResponder *parentResponder = self;
+  while ((parentResponder = parentResponder.nextResponder)) {
+    if ([parentResponder isKindOfClass:UIViewController.class]) {
+      return (UIViewController *)parentResponder;
+    }
+  }
+  return nil;
+}
+
+- (BOOL)startSigning {
+  // Programmatically show the signature view controller
+  PSPDFSignatureViewController *signatureController = [[PSPDFSignatureViewController alloc] init];
+  signatureController.naturalDrawingEnabled = YES;
+  signatureController.delegate = self;
+  PSPDFNavigationController *signatureContainer = [[PSPDFNavigationController alloc] initWithRootViewController:signatureController];
+  [_pdfController presentViewController:signatureContainer animated:YES completion:NULL];
+
+  return YES;
+}
+
+#pragma mark - PSPDFSignatureViewControllerDelegate
+
+// Sign all pages example
+- (void)signatureViewControllerDidFinish:(PSPDFSignatureViewController *)signatureController withSigner:(nullable PSPDFSigner *)signer shouldSaveSignature:(BOOL)shouldSaveSignature {
+    [signatureController dismissViewControllerAnimated:YES completion:^{
+      NSURL *samplesURL = [NSBundle.mainBundle.resourceURL URLByAppendingPathComponent:@"PDFs"];
+      NSURL *p12URL = [samplesURL URLByAppendingPathComponent:@"JohnAppleseed.p12"];
+
+      NSData *p12data = [NSData dataWithContentsOfURL:p12URL];
+      NSAssert(p12data, @"Error reading p12 data from %@", p12URL);
+      PSPDFPKCS12 *p12 = [[PSPDFPKCS12 alloc] initWithData:p12data];
+      PSPDFPKCS12Signer *signer = [[PSPDFPKCS12Signer alloc] initWithDisplayName:@"John Appleseed" PKCS12:p12];
+      signer.reason = @"Contract agreement";
+      PSPDFSignatureManager *signatureManager = PSPDFKitGlobal.sharedInstance.signatureManager;
+      [signatureManager clearRegisteredSigners];
+      [signatureManager registerSigner:signer];
```

Comment:
We don't actually need to register the signer, since we use it directly to sign the form element, instead of using the signature view.